### PR TITLE
Use conda for all actions

### PR DESCRIPTION
**Pull request by jorgensd**
- Optional docker-container based on conda environment available
- Use mamba on CI
- Use environment.yml for binder.

### CODE REVIEW

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -21,12 +20,9 @@ jobs:
...
 
-      - name: Install book deps
-        run:
-          python3 -m pip install -r requirements.txt
-
       - name: Build the book
-        run: 
+        run: |
+          conda activate mpi
           jupyter-book build . -W
```
Comment (minrk) [inline .github/workflows/build_docs.yml]: You can avoid activation troubles with `conda run`:

```suggestion
          conda run -n mpi jupyter-book build . -W
```

File: .github/workflows/build_docs.yml [hunk 2]
```diff
@@ -21,14 +20,9 @@ jobs:
...
 
-      - name: Install book deps
-        run:
-          python3 -m pip install -r requirements.txt
-
       - name: Build the book
         run: 
-          jupyter-book build . -W
-
+           conda run -n mpi-tutorial jupyter-book build . -W
```
Comment (minrk) [inline .github/workflows/build_docs.yml]: Oops, it's called `mpi` (or rename the env above)

```suggestion
           conda run -n mpi jupyter-book build . -W
```

Comment (jorgensd) [reply .github/workflows/build_docs.yml]: I've been changing it back and forth, as  it seems like there has to be an env naming consistency.

Comment (jorgensd) [reply .github/workflows/build_docs.yml]: Sorry about that.

File: .github/actions/install-dependencies/action.yml [hunk 3]
```diff
@@ -1,22 +1,40 @@
...
+          ln -s $CONDA/envs /opt/conda/envs
+        fi
+        echo CONDA=$CONDA
+        mamba env list
+        
+    - name: Environment cache
+      uses: actions/cache@v3
+      with:
+        path: /opt/conda/envs/mpi-tutorial
+        key: env-conda-mpi-tutorial-${{ inputs.python-version }}-${{ hashFiles('.github/actions/install-dependencies/*') }}
```
Comment (minrk) [inline .github/actions/install-dependencies/action.yml]: cache key should include the environment.yml, presumably

Comment (jorgensd) [reply .github/actions/install-dependencies/action.yml]: Do you have a nice way of generate a hash based on a text file?

**Pull request merged**